src/components/SearchForm/SearchForm.js

```diff
@@ -1,19 +1,32 @@
 import { useState } from "react";
 import Preloader from "../Preloader/Preloader";
+import api from "../../utils/NewsApi";
 
-export default function SearchForm({ onSearch, onLoading, loadingStatus: isLoading }) {
+export default function SearchForm({ onError, onSearch, onLoading, loadingStatus: isLoading }) {
     const [query, setQuery] = useState("");
+    const [errorText, setErrorText] = useState("");
     const [lastQuery, setLastQuery] = useState("");
 
     const handleFormSubmit = (e) => {
         e.preventDefault();
+        if (query.length < 2) {
+            setErrorText("Please enter a keyword (minimum 2 characters)");
+        }
         if (query.length > 2 && query !== lastQuery) {
+            setErrorText("");
             onLoading(true);
-            setTimeout(() => {
-                onSearch(query);
-                setLastQuery(query);
-                onLoading(false);
-            }, 1500);
+            api.getNews({query})
```

Comment:
**Might be better**


* it's better to make api requests only in `App` because `Login` and `Register` should not know how to do it. They should be independent from such logics